src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -878,6 +878,18 @@ export class BundleDataClient {
                   isDefined(v3RelayHashes[relayDataHash].deposit),
                   "Deposit should exist in relay hash dictionary."
                 );
+                // If the fill's repayment address is not a valid EVM address and the repayment chain is an EVM chain, the fill is invalid.
+                if (chainIsEvm(fill.repaymentChainId) && !isValidEvmAddress(fill.relayer)) {
```

Comment:
Can we refactor this check into a helper function since its code is basically re-used in the historical deposit case?

Reply:
Also, do you think we can move this code into `updateBundleFillsV3`? We can still update invalid fills from there

Reply:
Refactored here: https://github.com/across-protocol/sdk/pull/842/commits/f648d61d360d0c80b0d3e4c3771fb1e15fc1a987